src/Rector/Contrib/Symfony/FrameworkBundle/AbstractToConstructorInjectionRector.php

```diff
@@ -78,4 +77,27 @@ public function refactor(Node $methodCallNode): ?Node
 
         return $this->propertyFetchNodeFactory->createLocalWithPropertyName($propertyName);
     }
+
+    /**
+     * @param MethodCall $methodCallNode
+     */
+    private function serviceType(Node $methodCallNode): ?string
+    {
+        $argument = $methodCallNode->args[0]->value;
+
+        if ($argument instanceof String_) {
+            $serviceName = $argument->value;
+            return $this->serviceTypeForNameProvider->provideTypeForName($serviceName);
+        }
+
+        if (! $argument instanceof ClassConstFetch) {
+            return null;
+        }
+
+        if ($argument->class instanceof Name) {
+            return $argument->class->getAttribute(Attribute::RESOLVED_NAME)->toString();
+        }
+
+        return null;
```

Comment:
Can't we move the `if ($argument->class instanceof Name) {` before the `if (! $argument instanceof ClassConstFetch) {` to clean up an `if`?

Reply:
PHPStan wont let as $argument->class may not be available.